test/scenarios/requires-type-checking/tslint.json

```diff
@@ -0,0 +1,5 @@
+{
```

Comment:
I think we need to add `tsconfig.json` as a config option, just like the CLI `tslint --project <filepath>`, and add one to this test scenario

Reply:
I expect the code here would not work when you're type checking a file, but you have definitions defined in a .d.ts file somewhere that wasn't passed to createProgram.

Here's how they called createProgram in tslint: https://github.com/palantir/tslint/blob/f22a3af28f2e3429ca223b0438b4a9b5b5ce0d62/src/linter.ts

I notice that in tslint they call createProgram once for all files, and create a single linter. Here it looks like grunt-tslint creates a linter and program for every file.